### PR TITLE
Issue 1 create an entry

### PR DESCRIPTION
**Pull request by tangm28**
Fixed the bugs in issue 1.

creating-entry-michael_tang

### CODE REVIEW

File: js/data.js [hunk 1]
```diff
@@ -1,8 +1,26 @@
...
   view: 'entry-form',
   entries: [],
   editing: null,
   nextEntryId: 1
 };
+
+function loadData(event) {
+  previousEntries = localStorage.getItem('entries-local-storage');
+  if (previousEntries !== null) {
+    data.entries = JSON.parse(previousEntries).entries;
```
Comment (BrettAlbright) [inline js/data.js]: in here, you just want to set `data` to be equal to what is in local storage. you don't need to access properties or increment values. 

The value incrementation for the `nextEntryId` should be done after the new entry is unshifted to the entries array

File: js/data.js [hunk 2]
```diff
@@ -1,8 +1,26 @@
...
+  }
+}
+
+function saveDataB4Unload(event) {
+  if (data.entries.length > 0) {
+    localStorage.setItem('entries-local-storage', JSON.stringify(data));
+  }
+}
+
+window.addEventListener('load', loadData);
```
Comment (BrettAlbright) [inline js/data.js]: you don't need this here, you can just check local storage when the file is read.

**Review by Zhiwutian — CHANGES_REQUESTED** [review]
almost there!

File: js/data.js (hunk 2 above)
Comment (tangm28) [reply js/data.js]: is there a reason why we would want to code to run when the script is loaded vs when the page is loaded?

**Review by Zhiwutian — APPROVED** [review]
👍🏼

**Pull request merged**